generic3g/actions/AccumulatorActionInterface.F90

```diff
@@ -35,33 +38,35 @@ module mapl3g_AccumulatorActionInterface
    logical function accumulation_type_is_valid(acctype) result(lval)
       character(len=*), optional, intent(in) :: acctype
 
-      lval = present(acctype)
-      if(lval) lval = any(ACCUMULATION_TYPES == acctype)
+      lval = .FALSE.
+      if(.not. present(acctype)) return
+      lval = any(ACCUMULATION_TYPES == acctype)
 
    end function accumulation_type_is_valid
 
    subroutine get_accumulator_action(accumulation_type, typekind, action, rc)
       character(len=*), intent(in) :: accumulation_type
       type(ESMF_TypeKind_Flag), intent(in) :: typekind
-      class(ExtensionAction), allocatable, intent(out) :: action
+      class(ExtensionAction), allocatable, intent(inout) :: action
       integer, optional, intent(out) :: rc 
 
       integer :: status
 
       if(typekind /= ESMF_TYPEKIND_R4) then
          _FAIL('Unsupported typekind')
       end if
-      _ASSERT(accumulation_type_is_valid(accumulation_type), 'Unsupported AccumulatorAction')
 
       select case(accumulation_type)
       case (SIMPLE_ACCUMULATION)
-         action = AccumulatorAction(typekind)
+         allocate(action, source=AccumulatorAction(typekind))
       case (MEAN_ACCUMULATION)
-         action = MeanAction(typekind)
+         allocate(action, source=MeanAction(typekind))
       case (MAX_ACCUMULATION)
-         action = MaxAction(typekind)
+         allocate(action, source=MaxAction(typekind))
       case (MIN_ACCUMULATION)
-         action = MinAction(typekind)
+         allocate(action, source=MinAction(typekind))
+      case (INSTANTANEOUS)
+         _FAIL('No AccumulatorAction for instantaneous.')
```

Comment:
We do have `NullAction()`  if you want to use it here.  (Mostly relevant because of my comment about INOUT above.)

Reply:
Done